tests/ci_visibility/util.py

```diff
@@ -118,3 +119,35 @@ def _mock_upload_git_metadata(obj, **kwargs):
         "ddtrace.internal.ci_visibility.recorder._do_request", side_effect=NotImplementedError
     ):
         yield
+
+
+_PYTEST_SNAPSHOT_GITLAB_CI_ENV_VARS = {
+    "GITLAB_CI": "true",
+    "CI_COMMIT_AUTHOR": "TestFirst TestLast <First.Last@testtest.com>",
+    "CI_COMMIT_MESSAGE": "test commit message",
+    "CI_COMMIT_REF_NAME": "test.brancn/test_name",
+    "CI_COMMIT_SHA": "c165eb71ef833b752783b5268f21521fd16f812a",
+    "CI_COMMIT_TIMESTAMP": "2024-09-10T10:11:13+01:00",
+    "CI_COMMIT_TAG": "v1.0.0",
```

Comment:
[Was the error we initially saw](https://gitlab.ddbuild.io/DataDog/apm-reliability/dd-trace-py/-/jobs/633802338) because we were inheriting values like this from `os.env`, and here we're statically setting it, which should fix it?

Reply:
Exactly, yes, so a lot of the changes I'm making is to "drop" into a predictable, unchanging env when the tests run.

Unfortunately there's extra goofiness in some places because, when certain env vars are defined, then we don't use `git` to get other bits of info (eg: commit message, hashes, service, etc), so we can't only do one or the other. We have to have some tests with the GitLab env vars, and some without (which is also why I'm introducing the "clean env" boolean to `override_env()`.